plugins/woocommerce/client/blocks/assets/js/base/context/providers/cart-checkout/checkout-events/test/index.tsx

```diff
@@ -3,30 +3,97 @@
  */
 import { act, render } from '@testing-library/react';
 import { dispatch } from '@wordpress/data';
-import { checkoutStore } from '@woocommerce/block-data';
+import { checkoutStore, paymentStore } from '@woocommerce/block-data';
 import { checkoutEvents } from '@woocommerce/blocks-checkout-events';
 
 /**
  * Internal dependencies
  */
 import { CheckoutEventsProvider } from '../index';
 
+// Mock the registry functions
+jest.mock( '@woocommerce/blocks-registry', () => ( {
+	getPaymentMethods: jest.fn( () => ( {} ) ),
+	getExpressPaymentMethods: jest.fn( () => ( {
+		stripe: {
+			name: 'stripe',
+			title: 'Stripe',
+			description: 'Pay with Stripe',
+			gatewayId: 'stripe',
+			supports: {
+				style: [ 'height', 'borderRadius' ],
+			},
+		},
+		paypal: {
+			name: 'paypal',
+			title: 'PayPal',
+			description: 'Pay with PayPal',
+			gatewayId: 'paypal',
+			supports: {
+				style: [],
+			},
+		},
+	} ) ),
+} ) );
+
 describe( 'CheckoutEventsContext', () => {
+	let mockSetRegisteredExpressPaymentMethods: jest.Mock;
+
 	beforeEach( () => {
+		jest.clearAllMocks();
 		dispatch( checkoutStore ).__internalSetIdle();
+
+		// Mock the payment store dispatch action
+		mockSetRegisteredExpressPaymentMethods = jest.fn();
+		jest.spyOn(
+			dispatch( paymentStore ),
+			'__internalSetRegisteredExpressPaymentMethods'
+		).mockImplementation( mockSetRegisteredExpressPaymentMethods );
+	} );
+
+	it( '__internalSetRegisteredExpressPaymentMethods is called when component renders', () => {
+		render(
+			<CheckoutEventsProvider redirectUrl="local">
+				{ ( <div /> ) as unknown as React.ReactChildren }
+			</CheckoutEventsProvider>
+		);
+
+		expect( mockSetRegisteredExpressPaymentMethods ).toHaveBeenCalledWith( {
+			stripe: {
+				name: 'stripe',
+				title: 'Stripe',
+				description: 'Pay with Stripe',
+				gatewayId: 'stripe',
+				supportsStyle: [ 'height', 'borderRadius' ],
+			},
+			paypal: {
+				name: 'paypal',
+				title: 'PayPal',
+				description: 'Pay with PayPal',
+				gatewayId: 'paypal',
+				supportsStyle: [],
+			},
+		} );
 	} );
+
 	it( 'onCheckoutValidation observers are called when the checkout is in the "beforeProcessing" state', async () => {
 		const callback = jest.fn();
 		const callback2 = jest.fn();
 		checkoutEvents.onCheckoutValidation( callback );
 		checkoutEvents.onCheckoutValidation( callback2 );
 		const { rerender } = render(
-			<CheckoutEventsProvider redirectUrl="local" />
+			<CheckoutEventsProvider redirectUrl="local">
+				{ ( <div /> ) as unknown as React.ReactChildren }
```

Comment:
If you need to do this, I think something is broken with `CheckoutEventsProvider` types. I think changing the type there is a better option. Go with:
```ts
children?: React.ReactNode;
```
And say goodbye to problems 😊